api/src/services/files.ts

```diff
@@ -343,3 +347,66 @@ export class FilesService extends ItemsService {
 		return keys;
 	}
 }
+
+function decompressResponse(stream: Readable, headers: AxiosResponse['headers']) {
+	const contentEncoding = (headers['content-encoding'] || '').toLowerCase();
+
+	if (!['gzip', 'deflate', 'br'].includes(contentEncoding)) {
+		return stream;
+	}
+
+	let isEmpty = true;
+
+	function handleContentEncoding(data: any) {
+		const decompressStream =
+			// eslint-disable-next-line no-nested-ternary
+			contentEncoding === 'br'
+				? zlib.createBrotliDecompress()
+				: contentEncoding === 'deflate' && data.length > 0 && (data[0] & 0x08) === 0
```

Comment:
What exactly is the purpose of `0x08` here? Can we move this magic value to a constant for some added context? And should we use just plain `8` instead of the hex notation ```8 === 0x08```

Reply:
It's the header indicating compression with `deflate` algorithm. I've cleaned up the code 👍 

I prefer `0x08` in this case, because I feel like it's more obvious we're doing bitwise operation that way.

Reply:
beautiful, love the abstraction to its own function :) a whole lot clearer what the if statement does now